library/src/main/java/com/novoda/downloadmanager/lib/DownloadNotifier.java

```diff
@@ -422,7 +422,7 @@ private void setSecondaryNotificationText(NotificationCompat.Builder builder, No
     }
 
     private void removeStaleTagsThatWereNotRenewed(SimpleArrayMap<String, Collection<DownloadBatch>> clustered) {
-        for (int i = 0, size = activeNotifs.size(); i < size; i++) {
+        for (int i = activeNotifs.size() - 1; i >= 0; i--) {
```

Comment:
It is because if you delete going forward, the next item is two items beyond because you just deleted the current item so the next one is at your current position.

But going backwards you always have a valid item when you delete the current one because you move to the left and that is the correct item at that position.